### PR TITLE
[ENT-863] Deactivate the ECU if SAPSF has marked them inactive.

### PR DESCRIPTION
**Pull request by UmanShahzad**
**Description:** If SAPSF tells us the ECU is inactive for whatever reason (e.g. if they have left the company they were associated with), we unlink them on our side too and skip them for future transmissions.

**JIRA:** https://openedx.atlassian.net/browse/ENT-863

**Merge deadline:** June 1st.

**Testing instructions:**

Test instructions TBD.

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @UmanShahzad! I've created [OSPR-2438](https://openedx.atlassian.net/browse/OSPR-2438) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).



If you like, you can add yourself to the [AUTHORS](https://github.com/edx/edx-enterprise/blob/master/AUTHORS)      file for this repo, though that isn't required.     Please see the [CONTRIBUTING](https://github.com/edx/edx-platform/blob/master/CONTRIBUTING.rst)     file for more information.

File: integrated_channels/integrated_channel/transmitters/learner_data.py [hunk 1]
```diff
@@ -81,23 +81,27 @@ def transmit(self, payload, **kwargs):
...
-                    body,
-                    sys_msg
-                )
+                self.transmission_error_handler(learner_data, request_exception)
 
             learner_data.status = str(code)
             learner_data.error_message = body if code >= 400 else ''
             learner_data.save()
+
+    def transmission_error_handler(self, learner_data, request_exception):
```
[the remark this answers is not in the document]
Comment (UmanShahzad) [reply integrated_channels/integrated_channel/transmitters/learner_data.py]: `handle_transmission_error` sounds good to me.

File: integrated_channels/sap_success_factors/transmitters/learner_data.py [hunk 2]
```diff
@@ -35,3 +40,22 @@ def transmit(self, payload, **kwargs):
         kwargs['model_name'] = 'SapSuccessFactorsLearnerDataTransmissionAudit'
         kwargs['remote_user_id'] = 'sapsf_user_id'
         super(SapSuccessFactorsLearnerTransmitter, self).transmit(payload, **kwargs)
+
+    def transmission_error_handler(self, learner_data, request_exception):
+        """Handle the case where the employee on SAPSF's side is marked as inactive."""
+        try:
+            sys_msg = request_exception.response.content
+            if 'user account is inactive' in sys_msg:
```
[the remark this answers is not in the document]
Comment (UmanShahzad) [reply integrated_channels/sap_success_factors/transmitters/learner_data.py]: :+1:

File: integrated_channels/sap_success_factors/transmitters/learner_data.py [hunk 3]
```diff
@@ -35,3 +40,22 @@ def transmit(self, payload, **kwargs):
...
+                ecu = EnterpriseCustomerUser.objects.get(
+                    enterprise_enrollments__id=learner_data.enterprise_course_enrollment_id)
+                ecu.active = False
+                ecu.save()
+                LOGGER.warning(
+                    '%s with ID %s and email %s is a former employee of %s '
```
[the remark this answers is not in the document]
Comment (UmanShahzad) [reply integrated_channels/sap_success_factors/transmitters/learner_data.py]: :+1:

**Comment by edx-webhook** [comment]
Although this pull request is already merged, I've created [OSPR-5530](https://openedx.atlassian.net/browse/OSPR-5530) so that we can track it in Jira.

There is nothing you have to do. No action is needed from your side.  Thanks again for your contribution.

**Pull request merged**